docs/rules/no-boolean-default.md

```diff
@@ -0,0 +1,39 @@
+# Prevents boolean defaults from being set (vue/no-boolean-default)
+
+The rule prevents Boolean props from having a default value.
+
+
+## :book: Rule Details
+The rule is to enforce the HTML standard of always defaulting boolean attributes to false.
+
+Examples of **incorrect** code for this rule:
+```js
+export default {
+  props: {
+    foo: {
+      type: Boolean,
+      default: true
+    }
+  }
+}
+```
+
+Examples of **correct** code for this rule:
+```js
+export default {
+  props: {
+    foo: Boolean
+  }
+}
+```
+
+
+## :wrench: Options
+- `'constructor'` (default) enforces that Boolean props must use the constructor.
```

Comment:
There is a separate rule that checks whether prop type is a constructor, so I don't think we need it here. Perhaps the only additional setting we need here is `allowFalse`?